### PR TITLE
Hide and Edit Constraints and have the ConstraintView respond to changes

### PR DESCRIPTION
**Pull request by KamilRosiak**
1. We can edit Constraints in the ConstraintView with a double click or via context menu.
2. The Constraints under the diagram automatically disappear when the ConstraintView at the buttom is open and appears when you switch to different View at the buttom / close the ConstraintView
3. The ConstraintView adapts when the FeatureModelEditor changes or Constraints get added/edited/deleted in the Editor

### CODE REVIEW

File: plugins/de.ovgu.featureide.fm.ui/src/de/ovgu/featureide/fm/ui/utils/FeatureModelUtil.java [hunk 1]
```diff
@@ -33,28 +33,38 @@
  */
 public class FeatureModelUtil {
 	/**
+<<<<<<< HEAD
 	 * This method returns the active feature model editor if available.
```
Comment (Indycate) [inline plugins/de.ovgu.featureide.fm.ui/src/de/ovgu/featureide/fm/ui/utils/FeatureModelUtil.java]: check line 36(37) and 40

**Comment by KamilRosiak** [comment]
We did, it's fixed
@PaulWestphal

File: plugins/de.ovgu.featureide.fm.ui/src/de/ovgu/featureide/fm/ui/views/constraintview/actions/EditConstraintAction.java [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+import org.eclipse.jface.viewers.IStructuredSelection;
+import org.eclipse.jface.viewers.TreeViewer;
+
+import de.ovgu.featureide.fm.core.base.IConstraint;
+import de.ovgu.featureide.fm.core.base.IFeatureModel;
+import de.ovgu.featureide.fm.ui.FMUIPlugin;
+import de.ovgu.featureide.fm.ui.editors.featuremodel.actions.AbstractConstraintEditorAction;
+
+/**
+ * TODO description
```
Comment (PaulWestphal) [inline plugins/de.ovgu.featureide.fm.ui/src/de/ovgu/featureide/fm/ui/views/constraintview/actions/EditConstraintAction.java]: no

File: plugins/de.ovgu.featureide.fm.ui/src/de/ovgu/featureide/fm/ui/views/constraintview/actions/EditConstraintAction.java [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+ *
+ * @author "Rosiak Kamil"
+ */
+public class EditConstraintAction extends AbstractConstraintEditorAction {
```
Comment (PaulWestphal) [inline plugins/de.ovgu.featureide.fm.ui/src/de/ovgu/featureide/fm/ui/views/constraintview/actions/EditConstraintAction.java]: Having two classes with the same name is confusing. The class name should make it clear that this is an action for the Constraint View.

File: plugins/de.ovgu.featureide.fm.ui/src/de/ovgu/featureide/fm/ui/views/constraintview/view/ConstraintView.java [hunk 4]
```diff
@@ -21,109 +21,120 @@
...
-		final TableViewerColumn constraintViewerColumn = new TableViewerColumn(viewer, SWT.NONE);
-		final TableColumn constraintColumn = constraintViewerColumn.getColumn();
+	/**
+	 * Adding the columns with topics to the tree viewer
+	 */
+	private void addColumns(TreeViewer viewer) {
+		constraintColumn = new TreeColumn(viewer.getTree(), SWT.LEFT);
+		constraintColumn.setResizable(true);
+		constraintColumn.setMoveable(true);
+		constraintColumn.setWidth(800);
```
Comment (PaulWestphal) [inline plugins/de.ovgu.featureide.fm.ui/src/de/ovgu/featureide/fm/ui/views/constraintview/view/ConstraintView.java:131]: Use a final int

File: plugins/de.ovgu.featureide.fm.ui/src/de/ovgu/featureide/fm/ui/views/constraintview/view/ConstraintView.java [hunk 5]
```diff
@@ -21,109 +21,120 @@
...
 		constraintColumn.setText(CONSTRAINT_HEADER);
-		addConstraintColumnProvider(constraintViewerColumn);
 
-		final TableViewerColumn descriptionViewerColumn = new TableViewerColumn(viewer, SWT.NONE);
-		final TableColumn descriptionColumn = descriptionViewerColumn.getColumn();
+		descriptionColumn = new TreeColumn(viewer.getTree(), SWT.LEFT);
+		descriptionColumn.setResizable(true);
+		descriptionColumn.setMoveable(true);
+		descriptionColumn.setWidth(200);
```
Comment (PaulWestphal) [inline plugins/de.ovgu.featureide.fm.ui/src/de/ovgu/featureide/fm/ui/views/constraintview/view/ConstraintView.java:137]: final int

**Review by PaulWestphal — COMMENTED** [review]
EditConstraintAction needs to be renamed.

**Review by PaulWestphal — APPROVED** [review]

**Comment by tthuem** [comment]
This PR contains numerous changes to files not being necessary. For examples, line breaks are changed to files not touched otherwise.

**Pull request merged**